Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+
+        return issues
+
+    def PyInit(self):
+
+        self.declareProperty(MultipleFileProperty('Run', extensions=['nxs']),
+                             doc='File path of run(s).')
+
+        options = ['Absorber', 'Beam', 'Transmission', 'Container', 'Quartz', 'Vanadium', 'Sample']
+
+        self.declareProperty(name='ProcessAs',
+                             defaultValue='Sample',
+                             validator=StringListValidator(options),
+                             doc='Choose the process type.')
+
+        self.declareProperty(WorkspaceGroupProperty('OutputWorkspace', '',
+                                                    direction=Direction.Output,
+                                                    optional=PropertyMode.Optional),
+                             doc='The output workspace based on the value of ProcessAs.')
+
+        absorber = EnabledWhenProperty('ProcessAs', PropertyCriterion.IsEqualTo, 'Absorber')
+
+        beam = EnabledWhenProperty('ProcessAs', PropertyCriterion.IsEqualTo, 'Beam')
+
+        container = EnabledWhenProperty('ProcessAs', PropertyCriterion.IsEqualTo, 'Container')
+
+        sample = EnabledWhenProperty('ProcessAs', PropertyCriterion.IsEqualTo, 'Sample')
+
+        quartz = EnabledWhenProperty('ProcessAs', PropertyCriterion.IsEqualTo, 'Quartz')
+
+        transmission = EnabledWhenProperty('ProcessAs', PropertyCriterion.IsEqualTo, 'Transmission')
+
+        vanadium = EnabledWhenProperty('ProcessAs', PropertyCriterion.IsEqualTo, 'Vanadium')
+
+        reduction = EnabledWhenProperty(quartz, EnabledWhenProperty(vanadium, sample, LogicOperator.Or), LogicOperator.Or)
+
+        scan = EnabledWhenProperty(reduction, EnabledWhenProperty(absorber, container, LogicOperator.Or), LogicOperator.Or)
+
+        self.declareProperty(WorkspaceGroupProperty('AbsorberInputWorkspace', '',
+                                                    direction=Direction.Input,
+                                                    optional=PropertyMode.Optional),
+                             doc='The name of the absorber workspace.')
+
+        self.setPropertySettings('AbsorberInputWorkspace',
+                                 EnabledWhenProperty(quartz,
+                                                     EnabledWhenProperty(vanadium, sample, LogicOperator.Or),
+                                                     LogicOperator.Or))
+
+        self.declareProperty(MatrixWorkspaceProperty('BeamInputWorkspace', '',
+                                                     direction=Direction.Input,
+                                                     optional=PropertyMode.Optional),
+                             doc='The name of the empty beam input workspace.')
+
+        self.setPropertySettings('BeamInputWorkspace', transmission)
+
+        self.declareProperty(MatrixWorkspaceProperty('CadmiumTransmissionInputWorkspace', '',
```

Comment:
You name the process as Absorber, but the property Cadmium. Rename this to Absorber too. Btw, what is cadmium transmission? is it not 0?

Reply:
Cadmium transmission is in the order of 1%.